src/librustc_typeck/astconv.rs

```diff
@@ -2358,10 +2358,10 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     break;
                 }
             }
-            for binding in segment.generic_args().bindings {
+
+            if let Some(binding) = segment.generic_args().bindings.get(0) {
```

Comment:
```suggestion
            if let [binding, ..] = segment.generic_args().bindings {
```
Although cc @varkor re. whether this weird `break` was correct in the first place. I think this deserves a comment at any rate.